packages/wallet/src/account.ts

```diff
@@ -243,14 +243,41 @@ export class Account extends AbstractAccount {
     const params: TxParamsType = { gasLimit: maxGasPerTx, ...txParams };
     const request = new ScriptTransactionRequest(params);
     request.addCoinOutput(destination, amount, assetId);
-
     const { maxFee, requiredQuantities } = await this.provider.getTransactionCost(request);
 
     await this.fund(request, requiredQuantities, maxFee);
 
     return this.sendTransaction(request);
   }
 
+  /**
+   * A helper that prepares a transaction request for calculating the transaction ID.
+   *
+   * @param destination - The address of the destination.
+   * @param amount - The amount of coins to transfer.
+   * @param assetId - The asset ID of the coins to transfer.
+   * @param txParams - The transaction parameters (gasLimit, gasPrice, maturity).
+   * @returns A promise that resolves to the prepared transaction request.
+   */
+  protected async prepareTransferTxRequestForIdCalculation(
+    /** Address of the destination */
+    destination: AbstractAddress,
+    /** Amount of coins */
+    amount: BigNumberish,
+    /** Asset ID of coins */
+    assetId: BytesLike = BaseAssetId,
+    /** Tx Params */
+    txParams: TxParamsType = {}
+  ): Promise<TransactionRequest> {
+    const { maxGasPerTx } = this.provider.getGasConfig();
+    const params: TxParamsType = { gasLimit: maxGasPerTx, ...txParams };
+    const request = new ScriptTransactionRequest(params);
+    request.addCoinOutput(destination, amount, assetId);
+    const { maxFee, requiredQuantities } = await this.provider.getTransactionCost(request);
+    await this.fund(request, requiredQuantities, maxFee);
+    return request;
+  }
+
```

Comment:
@Dhaiwat10 Could the `transfer` method reuse this one?

If yes, we could rename the method to `prepareTransferTxRequest` and have something like:

```ts
 async transfer(
    // ...
  ): Promise<TransactionResponse> {
    const txRequest = this.prepareTransferTxRequest(...);
    return this.sendTransaction(txRequest);
  }
```

Reply:
Done @arboleya 🤝